File: day11/part1-2.py
import sys

try:
    filename = sys.argv[1] + ".txt"
except IndexError:
    filename = "input.txt"

nums = open(filename).read().split()
nums = list(map(int, nums))

iterations = 25

for it in range(iterations):
    i = 0
    print("Iteration", it)
    while i < len(nums):
        if nums[i] == 0: nums[i] = 1
        elif len(str(nums[i])) % 2 == 0:
            string = str(nums[i])
            length = int(len(string) / 2)
            first = string[:length]
            last = string[length:]
            
            nums.insert(i, int(first))
            i += 1
            nums[i] = int(last)
        else:
            nums[i] *= 2024
        i += 1
    #print(f"iteration {it}: {" ".join(list(map(str, nums)))}")

print("Part 1:", len(nums))

File: blinking-pebbles-day11/part1-2.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;

class Day11 {
    public static void main(String[] args) {
        String filename = "input.txt";
        Scanner scn = null;
        try {
            scn = new Scanner(new File(filename));
        } catch (FileNotFoundException e) {
            System.out.println("ei tiedostoo");
            System.out.println(e.getMessage());
            System.exit(1);
        }

        List<Long> nums = new ArrayList<>();

        String line = scn.nextLine();
        scn.close();
        for (String str : line.split(" ")) {
            nums.add(Long.parseLong(str));
        }
        
        int iterations = 75;

        for (long it = 0; it < iterations; it++) {
            System.out.println("Iteration: " + it);
            // System.out.println(nums.toString());
            int i = 0;
            while (i < nums.size()) {
                long value = nums.get(i);
                if (value == 0) nums.set(i, (long) 1);
                else if (String.valueOf(value).length() % 2 == 0) {
                    String str = String.valueOf(value);
                    int length = str.length() / 2;
                    String first = str.substring(0, length);
                    String last = str.substring(length, str.length());

                    nums.add(i, Long.parseLong(first));
                    i++;
                    nums.set(i, Long.parseLong(last));
                } else {
                    nums.set(i, value * 2024);
                }
                i++;
            }
        }

        System.out.println("Length: " + nums.size());
    }
}